### PR TITLE
Add Google TypeScript code formatter

### PR DESCRIPTION
**Pull request by michplunkett**
Part of ticket: https://github.com/Matico-Platform/matico/issues/211

## Overview

Added the [Google Type Script](https://github.com/google/gts) package for our linting and code styling.

### CODE REVIEW

**Comment by stuartlynn** [comment]
Any reason why Google TypeScript over Prettier? Happy to go either way, just wanted to know if there are added advantages for googles implementation. Have seen Prettier used in a bunch of projects but this is the first time I have seen Google Typescript so curious

**Comment by michplunkett** [comment]
1. I've also used `eslint`, which is what this formatter uses, a handful of times and haven't had any complaints about it.
2. I think it'd be fun to deviate from what I know (We used Prettier on a lot of previous projects I've been on) as it's fun to use new tools and I trust Google in regards to their technical reasoning.
3. Since we are also likely to use `cargo fmt` I didn't necessarily se the benefit of using Prettier as opposed to anything else.

All that being said, I'm happy to roll with Prettier if you'd feel more comfortable with it.

**Comment by stuartlynn** [comment]
Sounds good, lets try it out. Just wanted to make sure I understood the reasoning.

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+console.log('Try npm run lint/fix!');
```
Comment (stuartlynn) [inline src/index.ts:1]: We might want to remove this. There probably shouldn't be a src dir at the root of the monorepo especially if it's just for testing stuff.

**Review by stuartlynn — APPROVED** [review]
This looks good. Feel free to merge in

**Comment by stuartlynn** [comment]
Just another quick note. This should be merged in to develop rather than main. That's our active development branch which we periodically pull in to main. We might want to change that soon though now we have three of us working on this

File: src/index.ts (hunk 1 above)
Comment (michplunkett) [reply src/index.ts:1]: I'll take care of this and reach out before any merges are done!

**Comment by michplunkett** [comment]
I removed the `src` directory and moved the branch from `main` to `develop`.

**Pull request merged**